pkg/kubelet/kubelet_pods.go

```diff
@@ -826,6 +826,19 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod
 			return "", err
 		}
 		return hostIPs[0].String(), nil
+	case "status.hostIPs":
+		if !utilfeature.DefaultFeatureGate.Enabled(features.PodHostIPs) {
```

Comment:
Don't check the gate here - pods that get through the API checks should work

Reply:
This deals with a pod that uses this feature and then turns it off, which will lead to this, and since the pod is already using this downward api we can't reject it, and we shouldn't give the value